api4/channel.go

```diff
@@ -1150,9 +1150,17 @@ func getChannelByNameForTeamName(c *Context, w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	if !c.App.SessionHasPermissionToChannel(*c.App.Session(), channel.Id, model.PERMISSION_READ_CHANNEL) {
-		c.Err = model.NewAppError("getChannelByNameForTeamName", "app.channel.get_by_name.missing.app_error", nil, "teamId="+channel.TeamId+", "+"name="+channel.Name+"", http.StatusNotFound)
-		return
+	if channel.Type == model.CHANNEL_OPEN {
+		if !c.App.SessionHasPermissionToTeam(*c.App.Session(), channel.TeamId, model.PERMISSION_READ_PUBLIC_CHANNEL) &&
+			!c.App.SessionHasPermissionToChannel(*c.App.Session(), channel.Id, model.PERMISSION_READ_CHANNEL) {
```

Comment:
Nit: It might be cleaner to refactor the permission bools to separate variables and then uplift the checks in the if statement itself.

```go
channelOk := c.App.SessionHasPermissionToChannel(*c.App.Session(), channel.Id, model.PERMISSION_READ_CHANNEL)
teamOk := c.App.SessionHasPermissionToTeam(*c.App.Session(), channel.TeamId, model.PERMISSION_READ_PUBLIC_CHANNEL)


if channel.Type == model.CHANNEL_OPEN && !channelOk && !teamOk {
   // throw error
} else if !channelOk {
  // throw error
}
```

Reply:
This suggested change improves readability a lot, so I've applied it. The if channelOk and teamOk check can't be uplifted since we want the `else` to trigger only if the channel type is not public, and not on either of the other failing checks.